docassemble/ALAutomatedTestingTests/al_set_up_testing.py

```diff
@@ -70,14 +108,48 @@ def update_github( self ):
   def get_github_info_from_repo_url( self ):
     """Use repo address to parse out owner name and repo name. Needs self.repo_url"""
     # Match either the actual URL or the clone HTTP or SSH URL
-    matches = re.match(r"^.+github.com(?:\/|:)([^\/]*)\/([^\/.]*)(?:\..{3})?", self.repo_url)
+    matches = re.match( r"^.*github.com(?:\/|:)([^\/]*)\/?([^\/.]*)?(?:\..{3})?", self.repo_url )
     if matches:
+      self.owner_name = matches.group(1)
       self.repo_name = matches.group(2)
     else:
-      self.repo_name = None
+      self.owner_name = ''
+      self.repo_name = ''
+      # Show error
+      error = ErrorLikeObject( message='GitHub URL', details=self.github_url_error )
+      self.errors.append( error )
+      log( error.__dict__, 'console' )
       
     return self
   
+  def get_free_branch_name( self ):
+    """Return an object with two values:
+    - found_free_name: Whether an appropriate branch name was free
+    - branch_name: The last branch name that was tried"""
+    # Get all branches
+    all_branches = self.repo.get_branches()
+    
+    # Control how many times the loop will run
+    count = 0
+    max_count = 20
+    found_free_name = False  # Start the loop off correctly
+    branch_name_base = self.default_branch_name
+    branch_name = branch_name_base
+    # Try every permitted new branch name until one is free
+    while ( not found_free_name and count < max_count ):
+      count += 1  # Ensure no infinite loop
+      
+      found_free_name = True  # The name is free until proven otherwise
+      for existing_branch in all_branches:
+        existing_branch_name = existing_branch.name
+        # If branch already exists
+        if existing_branch_name == branch_name:
```

Comment:
I think we could make this comparison simpler. 

Part 1 (line 130): 
```python
all_branches = self.repo.get_branches()
all_branch_names = [b.name for b in all_branches]
```

Part 2 (here):
```python
if branch_name in all_branch_names:
    branch_name = branch_name_base + '_' + str( count )
else:
    found_free_name = True
```